### PR TITLE
Added  Custom repo scenario

### PR DESCRIPTION
**Pull request by vijay8451**
- Added Custom repo scenario.
- Covered ScenarioBug1429201 in it.
- Fixed https://github.com/SatelliteQE/satellite6-upgrade/issues/309.
- Test results:
```
tests/upgrades/test_repository.py::Scenario_custom_repo_check::test_pre_scenario_custom_repo_check 2019-04-17 12:38:54 
PASSED2019-04-17 12:43:45 - robottelo - INFO - Started tearDownClass: tests.upgrades.test_repository/Scenario_custom_repo_check


tests/upgrades/test_repository.py::Scenario_custom_repo_check::test_post_scenario_custom_repo_check 2019-04-17 12:46:01 
PASSED2019-04-17 12:47:14 - robottelo - INFO - Started tearDownClass: tests.upgrades.test_repository/Scenario_custom_repo_check

```

### CODE REVIEW

File: tests/upgrades/test_repository.py [hunk 1]
```diff
@@ -89,3 +100,241 @@ def test_post_repository_scenario_upstream_authorization(self):
...
+    Test Steps:
+
+            1. Before Satellite upgrade.
+            2. Create new Organization and Location.
+            3. Create Product, custom repo, cv.
+            4. Create activation key and add subscription in it.
+            5. Create a content host, register and install package on it.
+            6. Upgrade Satellite.
+            7. Remove Old package and add new package into custom repo.
+            8. Sync repo , publish new version of cv.
```
Comment (omkarkhatavkar) [inline tests/upgrades/test_repository.py]: extra space here **sync repo ,**

**Comment by omkarkhatavkar** [comment]
ACK

File: tests/upgrades/test_repository.py (hunk 1 above)
Comment (vijay8451) [reply tests/upgrades/test_repository.py]: removed the same in latest push.thanks

File: tests/upgrades/test_repository.py [hunk 2]
```diff
@@ -14,12 +14,23 @@
 
 :Upstream: No
 """
-from fabric.api import run
+from fabric.api import execute, run
+import os
```
Comment (jyejare) [inline tests/upgrades/test_repository.py]: import comes first then from, sort alphabetically.

File: tests/upgrades/test_repository.py [hunk 3]
```diff
@@ -89,3 +100,241 @@ def test_post_repository_scenario_upstream_authorization(self):
...
+        kwargs = {'host': self.docker_vm}
+        execute(docker_execute_command,
+                client_container_id,
+                'subscription-manager repos;yum clean all',
+                **kwargs)[self.docker_vm]
+        command = 'yum install -y {}'.format(rpm_name)
+        execute(docker_execute_command, client_container_id, command, **kwargs)[self.docker_vm]
+        self._check_package_installed(client_container_id, rpm_name)
+
+    def _host_status(self, client_container_name=None):
```
Comment (jyejare) [inline tests/upgrades/test_repository.py]: Why do we need function for this ? You can directly use it.

File: tests/upgrades/test_repository.py (hunk 2 above)
Comment (vijay8451) [reply tests/upgrades/test_repository.py]: done , thanks

File: tests/upgrades/test_repository.py (hunk 3 above)
Comment (vijay8451) [reply tests/upgrades/test_repository.py]: I am using same at mutiple time under  `_host_location_update` hence used in this way.

**Comment by jyejare** [comment]
ACK

**Pull request merged**